widget/hyperlink.go

```diff
@@ -0,0 +1,140 @@
+func NewHyperlink(text string, sUrl string) *Hyperlink {
+	// ignore the error here, we'll continue anyway
+	return NewHyperlinkWithStyle(text, sUrl, fyne.TextAlignLeading, fyne.TextStyle{})
+}
+
+// NewHyperlinkWithStyle creates a new layout widget with the set text content
+func NewHyperlinkWithStyle(text string, sUrl string, alignment fyne.TextAlign, style fyne.TextStyle) *Hyperlink {
+	hl := &Hyperlink{
+		Text:      text,
+		Url:       sUrl,
+		Alignment: alignment,
+		TextStyle: style,
+	}
+
+	return hl
+}
+
+// SetText sets the text of the hyperlink
+func (hl *Hyperlink) SetText(text string) {
+	hl.Text = text
+	hl.textProvider.SetText(text)
+	Renderer(hl).Refresh()
+}
+
+// SetUrl sets the URL of the hyperlink, taking in a string type
+func (hl *Hyperlink) SetUrl(sUrl string) {
+	hl.Url = sUrl
+}
+
+// SetUrl sets the URL of the hyperlink, taking in a Url type
+func (hl *Hyperlink) SetUrlFromUrl(uUrl *url.URL) {
+	hl.Url = uUrl.String()
+}
+
+// textAlign tells the rendering textProvider our alignment
+func (hl *Hyperlink) textAlign() fyne.TextAlign {
+	return hl.Alignment
+}
+
+// textStyle tells the rendering textProvider our style
+func (hl *Hyperlink) textStyle() fyne.TextStyle {
+	return hl.TextStyle
+}
+
+// textColor tells the rendering textProvider our color
+func (hl *Hyperlink) textColor() color.Color {
+	return theme.HyperlinkColor()
+}
+
+// password tells the rendering textProvider if we are a password field
+func (hl *Hyperlink) password() bool {
+	return false
+}
+
+// object returns the root object of the widget so it can be referenced
+func (hl *Hyperlink) object() fyne.Widget {
+	return hl
+}
+
+// OnMouseDown is called when a mouse down event is captured and triggers any change handler
+func (hl *Hyperlink) OnMouseDown(*fyne.MouseEvent) {
+	if hl.Url != "" {
+		open(hl.Url)
+	}
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to it's renderer
+func (hl *Hyperlink) CreateRenderer() fyne.WidgetRenderer {
+	hl.textProvider = newTextProvider(hl.Text, hl)
+	return hl.textProvider.CreateRenderer()
+}
+
+// Resize sets a new size for a widget.
+// Note this should not be used if the widget is being managed by a Layout within a Container.
+func (hl *Hyperlink) Resize(size fyne.Size) {
+	hl.resize(size, hl)
+}
+
+// Move the widget to a new position, relative to it's parent.
+// Note this should not be used if the widget is being managed by a Layout within a Container.
+func (hl *Hyperlink) Move(pos fyne.Position) {
+	hl.move(pos, hl)
+}
+
+// MinSize returns the smallest size this widget can shrink to
+func (hl *Hyperlink) MinSize() fyne.Size {
+	return hl.minSize(hl)
+}
+
+// Show this widget, if it was previously hidden
+func (hl *Hyperlink) Show() {
+	hl.show(hl)
+}
+
+// Hide this widget, if it was previously visible
+func (hl *Hyperlink) Hide() {
+	hl.hide(hl)
+}
+
+// taken from https://github.com/icza/gowut/blob/085680a418c9a92dcf2c72d48df2df3cf2a1e88f/gwu/server_start.go#L29
```

Comment:
!LICENSE WARNING!

We cannot accept code from a GPL or LGPL project in this, BSD licensed, project.

Thankfully this code chunk is not needed as the app provider already manages opening URLs

Reply:
Changed in 3fdc033528934b7e8ed483474c570a7f82cbcc9a